src/Mvc/Mvc.Razor/src/RazorPageActivator.cs

```diff
@@ -88,7 +87,7 @@ internal RazorPagePropertyActivator GetOrAddCacheEntry(IRazorPage page)
                 var modelType = providedModelType;
                 if (modelType == null)
                 {
-                    modelType = pageType.GetRuntimeProperty(ModelPropertyName)?.PropertyType;
+                    modelType = pageType.GetRuntimeProperty(ModelPropertyName)?.PropertyType!;
```

Comment:
The pattern of `?.Property!;` seems wrong, if we're going to use `!` we should commit all the way and not have the `?`

Reply:
Good call. The receiver allows for a null model. I should update the annotations.
```suggestion
                    modelType = pageType.GetRuntimeProperty(ModelPropertyName)?.PropertyType;
```